modules/database/src/ioc/db/dbScan.c

```diff
@@ -59,6 +59,35 @@ static epicsEventId startStopEvent;
 
 static volatile enum ctl scanCtl;
 
+/* epicsTime does not support addition/subtraction of epicsTimeStamp
+ * but we want to do integer arithmetic...
+ */
+typedef struct epicsPeriod {
+    epicsTimeStamp ts;
+} epicsPeriod;
```

Comment:
`nsec` is an `epicsUInt32` since that never needs to hold more than 1e9 (or ±2e9 after doing an addition/subtraction).

We might want to allow periods to be negative, so I see the main question as whether we need to be able to represent more than ±68 years (2**31 seconds) as a period. I'm going to say no, so use an `epicsInt32` for `secs` as well.

Reply:
OK, done.